File: app/src/main/java/com/levibostian/androidblanky/MainActivity.java
package com.levibostian.androidblanky;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import com.levibostian.androidblanky.fragment.MainFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        startupFragment();
    }

    private void startupFragment() {
        getSupportFragmentManager().beginTransaction()
                .add(R.id.fragment_container, MainFragment.newInstance())
                .commit();
    }

}

File: app/src/main/java/com/levibostian/androidblanky/view/ui/activity/MainActivity.kt
package com.levibostian.androidblanky.view.ui.activity

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.commit
import androidx.fragment.app.transaction
import androidx.navigation.NavController
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupWithNavController
import com.levibostian.androidblanky.R
import com.levibostian.androidblanky.service.DataDestroyer
import com.levibostian.androidblanky.service.event.LogoutUserEvent
import com.levibostian.androidblanky.service.logger.Logger
import com.levibostian.androidblanky.service.manager.UserManager
import com.levibostian.androidblanky.service.util.InstallReferrerProcessor
import com.levibostian.androidblanky.view.ui.fragment.MainFragment
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.activity_toolbar_fragment_container.*
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class MainActivity: BaseActivity() {

    private val navController: NavController
        get() = findNavController(R.id.nav_host_fragment)

    companion object {
        fun getIntent(context: Context): Intent = Intent(context, MainActivity::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        (toolbar as Toolbar).setupWithNavController(navController, AppBarConfiguration(navController.graph))
    }

}